### PR TITLE
feat(registry): Add plugin and cli registries to start with

### PR DESCRIPTION
**Pull request by plwalters**
Added initial registries for holding plugin and cli plugins

### CODE REVIEW

**Comment by plwalters** [comment]
@zewa666 @ahmedshuhel @joelcoxokc For your review - everything look cool?

File: cli-plugin-registry.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "plugins": [
+    {
+      "name": "sample",
+      "endpoint": "github",    // where to start looking?  github, npm, gist
+      "location": "/user/repo",   // where is it in the end point?
+      "libs": [
+        "examplelib"
+      ]    // what libs should be installed in the cli/lib directory?
```
Comment (ahmedshuhel) [inline cli-plugin-registry.json]: Could you please explain the `libs` section a little more with an example usage?

**Comment by EisenbergEffect** [comment]
Perhaps instead call it "cli" instead of lib? I think we'll actually have some other install targets in the future as well.

File: cli-plugin-registry.json (hunk 1 above)
Comment (plwalters) [inline cli-plugin-registry.json]: @ahmedshuhel The libs would be which cli plugins we want to register.

**Comment by EisenbergEffect** [comment]
I think it would be interesting to think of this as a way to automate a collection of installs. So, one entry might be able to say:
- install X Aurelia plugin with jspm
- install Y gulp plugin with npm
- install Z cli plugin with npm

So, that a package can kind of declare a collection of related installs. What do you all think?

**Comment by plwalters** [comment]
Agreed.  I've removed the cli-plugin-registry and added it as part of the main registry.  I think we can do the same with the gulp plugins as well when we get to that point.

**Pull request merged**